### DISCUSSION TITLE
dnsx sometimes eats piped inputs and doesn't pipe out

### DISCUSSION THREAD

**Discussion by CorruptedPixl** [projectdiscovery/dnsx#303] [Q&A]
This issue affects at least v1.1.1 (and at least v1.1.0 too)

### The issue:
When piping from subfinder to dnsx to httpx, sometimes dnsx doesn't process the piped input and quits.
dnsx also doesn't pipe out to httpx or following commands.
_See below for full output logs._

### Expected Behavior:
When running `subfinder -d projectdiscovery.io | dnsx -v | httpx -title` dnsx should process the subdomains found by subfinder and pipe the output to httpx.

### Steps To Reproduce:
I am running this on a fresh Ubuntu (22.04.1 LTS) machine with go, subfinder, dnsx and httpx installed.

Running `subfinder -d projectdiscovery.io | dnsx -v | httpx -title` should produce an output as followed:

_Correct output after a reboot_
![image](https://user-images.githubusercontent.com/38385453/193925527-802c09ca-5f95-4f34-8b02-f15ec7a6eb61.png)

However, at times, dnsx refuses to work and produces no output after subfinder finishes.

_Bugged output_
![image](https://user-images.githubusercontent.com/38385453/193924862-c3fe0a2e-6d3f-4d26-8b01-f42e56229a3b.png)

### Important notes
**Some friends and I have noted that rebooting the machine seems to fix this for a while.**

Now, since this (somehow) gets temporarily fixed after a reboot, I didn't want to create this issue as I thought it might've been a config error on my side. _However_ after chatting with friends, having them install the tools and trying it out, this seems to be a widespread issue and I felt that it was important enough to take a look at.

I have tried to gather as much info to be able to determine the place of this issue, but there are simply too many variables which I cannot check myself. This is why I hope some people here might have a different setup and can reproduce this issue on other environments to narrow down the issue.

**Comment by zevaryx** [comment]
<img width="641" alt="image" src="https://user-images.githubusercontent.com/57541873/193929943-4a15ef02-ed32-4cf2-9c4f-1fe6b660c113.png">

I'm one of the friends that tested it. Here's the system info of one of the systems I was able to reproduce it on. It may be related to multiple IPs available due to a variety of reasons (Docker in this case). This system was rebuilt very recently and only has Docker running on it

**Comment by CorruptedPixl** [comment]
I have done some more testing and can confirm that this bug occurs 100% of the time whenever I am connected to a vpn, specifically PIA in this case.

However, some of my friends weren't using a vpn and got the same issue so I'm not sure that's the cause.
_Note: on windows, dnsx works regardless if the vpn is active or not._

I'm thinking maybe the dns settings of the vpn interfere with dnsx on Linux?

**Comment by CorruptedPixl** [comment]
![image](https://user-images.githubusercontent.com/38385453/193931729-588a015b-76a0-4123-8c7c-524d8ea62909.png)
Figured I'd also drop my system info. As you can see, when connected to the vpn there are multiple interfaces available, when disconnected there is only one.

**Comment by Mzack9999** [comment]
Probably this depends on the UDP protocol drop due to VPN (see https://github.com/projectdiscovery/dnsx/issues/221). You can try with DOH if it's not possible to disable the VPN:
```console
$ echo scanme.sh | go run . -resolver doh:https://doh.opendns.com/dns-query:get -verbose
scanme.sh
```

**Comment by ehsandeep** [comment]
Moved to discussions as we unable to repro this!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.

**Comment by xhzeem** [comment]
I had same issue when using a VPN, once I closed it, it started working, there seem to be an issue with some VPNs
